test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeGenerator.java

```diff
@@ -419,6 +436,88 @@ private static void generateSetter(ClassWriter cw, String[] nameAndSigValue, Str
 		mv.visitEnd();
 	}
 	
+	private static void generateSetterStatic(ClassWriter cw, String[] nameAndSigValue, String className) {
+		boolean doubleDetected = false;
+		MethodVisitor mv = cw.visitMethod((ACC_PUBLIC | ACC_STATIC), "setStatic" + nameAndSigValue[0], "(Ljava/lang/Class;" + nameAndSigValue[1] + ")V", null, null);
+		mv.visitCode();
+		mv.visitVarInsn(ALOAD, 0);
+		switch (nameAndSigValue[1]) {
+		case "D":
+			mv.visitVarInsn(DLOAD, 1);
+			doubleDetected = true;
+			break;
+		case "I":
+		case "Z":
+		case "B":
+		case "C":
+		case "S":
+			mv.visitVarInsn(ILOAD, 1);
+			break;
+		case "F":
+			mv.visitVarInsn(FLOAD, 1);
+			break;
+		case "J":
+			mv.visitVarInsn(LLOAD, 1);
+			doubleDetected = true;
+			break;
+		default:
+			mv.visitVarInsn(ALOAD, 1);
+			break;
+		}
+		mv.visitFieldInsn(PUTSTATIC, className, nameAndSigValue[0], nameAndSigValue[1]);
+		mv.visitInsn(RETURN);
+		int maxStackAndLocals = (doubleDetected ? 3 : 2);
+		mv.visitMaxs(maxStackAndLocals, maxStackAndLocals);
```

Comment:
locals will be either 1 or 2 depending on whether the field is a single or double slot